compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenOps.td

```diff
@@ -102,6 +103,59 @@ def IREECodegen_ExtractStridedMetadataOp : Op<IREECodegen_Dialect, "extract_stri
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// SwizzleHintOp
+//===----------------------------------------------------------------------===//
+
+def IREECodegen_SwizzleHintOp : Op<IREECodegen_Dialect, "swizzle_hint", [
+    SameOperandsAndResultType, Pure]> {
+  let summary = "Hint to swizzle accesses according to an access pattern";
+  let description = [{
+    Optimization hint to swizzle all accesses to the memref this takes a view
+    of. This only affects reads/writes immediately consuming this operation and
+    is best effort. If the desired swizzling is not apparently possible, this
+    op will no-op. As a result, it should not be relied on for correctness.
+
+    Note that this only rewrites direct users. If there are any aliased loads
```

Comment:
What happens with subviews?

Reply:
Added a comment explaining, but any remaining subviews will cause swizzling to fail. The expectation is that we fold view-like things into loads/stores before resolving. If we find out that this is too restrictive, we could probably add support for subviews in the future. Maybe. Not sure if it really works with a subview because the swizzling might leave the region of memory covered by the subview.